healsparse/io_map_healpix.py

```diff
@@ -0,0 +1,52 @@
+import os
+import numpy as np
+
+from .fits_shim import _make_header, _write_healpix_filename
+
+
+def _write_map_healpix(hsp_map, filename, clobber=False):
+    """
+    Internal method to write a HealSparseMap with healpix EXPLICIT format.
+
+    Note that the coverage map is not persisted in this format, and only
+    floating point value maps are supported.
+
+    Parameters
+    ----------
+    hsp_map : `HealSparseMap`
+        HealSparseMap to write to a file.
+    filename : `str`
+        Name of file to save
+    clobber : `bool`, optional
+        Clobber existing file?  Default is False.
+
+    Raises
+    ------
+    RuntimeError if file exists and clobber is False.
+    NotImplementedError if persisting anything other than a floating point map.
+    """
+
+    if hsp_map.is_rec_array:
+        raise NotImplementedError("The healpix EXPLICIT output format is not supported for recarray maps.")
+    if hsp_map.is_integer_map:
+        raise NotImplementedError("The healpix EXPLICIT output format is not supported for integer maps.")
```

Comment:
I think this is perhaps being a bit too strict - most likely integer columns would be OK nowadays.  In particular, there is a hint in the [healpy 1.15 release notes](https://github.com/healpy/healpy/releases/tag/1.15.0) that non-float32 datatypes are now supported on I/O.

Reply:
Maybe allow it but print a "Some older tools may not be able to process integer-valued maps" warning?

Reply:
The release notes refer to the fact that it will now preserve 32-bit or 64-bit floats rather than have to force the user to know what it should be instead of incorrectly converting all the time (which was the annoying default behavior).
You can preserve integer maps in a healpix fits file, but healpy will (still) always convert ints to floats on read and converts all `BAD_DATA` to `hp.UNSEEN`.

But I'll let it save integer maps, which can be read in just fine by healsparse.  But I'll add a warning about healpy.

Reply:
I see; OK.  I didn't look deeply enou

Reply:
Some of my long-standing frustrations with healpy that we addressed with healsparse.  But it doesn't break, so I've added the feature.